### PR TITLE
PCC-174 derive Hash for Atom; update atom_generator

### PR DESCRIPTION
**Pull request by turboladen**

### CODE REVIEW

File: atom_generator/src/templates/classification.rs.hbs [hunk 1]
```diff
@@ -17,3 +17,32 @@ impl Default for Classification {
         Self::Si
     }
 }
+
+impl std::fmt::Display for Classification {
```
Comment (turboladen) [inline atom_generator/src/templates/classification.rs.hbs:21]: This had been added directly to `classification.rs`; the change here is just moving that to this (the codegen template)

**Review by kurtronshausen — APPROVED** [review]

File: atom_generator/src/templates/atom.rs.hbs [hunk 2]
```diff
@@ -14,7 +14,7 @@ use crate::{
 };
 use std::fmt;
 
-#[derive(Clone, Copy, Debug, Eq, PartialOrd, Ord)]
+#[derive(Clone, Copy, Debug, Eq, PartialOrd, Ord, Hash)]
```
Comment (turboladen) [inline atom_generator/src/templates/atom.rs.hbs:17]: Turns out deriving `Hash` but manually implementing `PartialEq` may not be the best: https://rust-lang.github.io/rust-clippy/master/index#/derived_hash_with_manual_eq. Probably should manually impl Hash.

**Comment by turboladen** [comment]
Working in handlebars+rust is ridiculous; I'm gonna see about switching to use https://docs.rs/quote/latest/quote/.

**Comment by kurtronshausen** [comment]
> Working in handlebars+rust is ridiculous; I'm gonna see about switching to use https://docs.rs/quote/latest/quote/.

I was wondering about that.  Is this not a proc-macro crate?

**Comment by turboladen** [comment]
> > Working in handlebars+rust is ridiculous; I'm gonna see about switching to use https://docs.rs/quote/latest/quote/.
> 
> I was wondering about that. Is this not a proc-macro crate?

Nope. Since the underlying unit definitions only change every couple of years, I made `atom_generator` an on-demand thing (you run it when you want to regenerate Rust defs from XML/TOML defs). I don't love it.

**Pull request closed without merging**